waltid-libraries/waltid-verifiable-credentials/src/commonMain/kotlin/id/walt/credentials/schemes/JwsSignatureScheme.kt

```diff
@@ -21,75 +22,79 @@ private val log = KotlinLogging.logger { }
+  }
 
-    object JwsOption {
-        const val SUBJECT = "sub"
-        const val ISSUER = "iss"
-        const val EXPIRATION = "exp"
-        const val NOT_BEFORE = "nbf"
-        const val VC_ID = "jti"
-        const val VC = "vc"
-    }
+  object JwsOption {
+    const val SUBJECT = "sub"
+    const val ISSUER = "iss"
+    const val EXPIRATION = "exp"
+    const val NOT_BEFORE = "nbf"
+    const val VC_ID = "jti"
+    const val VC = "vc"
+  }
 
-    /**
-     * args:
-     * - kid: Key ID
-     * - subjectDid: Holder DID
-     * - issuerDid: Issuer DID
-     */
-    @JvmBlocking
-    @JvmAsync
-    @JsPromise
-    @JsExport.Ignore
-    suspend fun sign(
-        data: JsonObject, key: Key,
-        /** Set additional options in the JWT header */
-        jwtHeaders: Map<String, String> = emptyMap(),
-        /** Set additional options in the JWT payload */
-        jwtOptions: Map<String, JsonElement> = emptyMap(),
-    ): String {
-        val payload = Json.encodeToString(
-            mapOf(
-                JwsOption.ISSUER to jwtOptions[JwsOption.ISSUER],
-                JwsOption.SUBJECT to jwtOptions[JwsOption.SUBJECT],
-                JwsOption.VC to data,
-                *(jwtOptions.entries.map { it.toPair() }.toTypedArray())
-            )
-        ).encodeToByteArray()
+  /**
+   * args:
+   * - kid: Key ID
+   * - subjectDid: Holder DID
+   * - issuerDid: Issuer DID
+   */
+  @JvmBlocking
+  @JvmAsync
+  @JsPromise
+  @JsExport.Ignore
+  suspend fun sign(
+    data: JsonObject, key: Key,
+    /** Set additional options in the JWT header */
+    jwtHeaders: Map<String, JsonElement> = emptyMap(),
+    /** Set additional options in the JWT payload */
+    jwtOptions: Map<String, JsonElement> = emptyMap(),
+  ): String {
+    val payload = Json.encodeToString(
+      mapOf(
+        JwsOption.ISSUER to jwtOptions[JwsOption.ISSUER],
+        JwsOption.SUBJECT to jwtOptions[JwsOption.SUBJECT],
+        JwsOption.VC to data,
+        *(jwtOptions.entries.map { it.toPair() }.toTypedArray())
+      )
+    ).encodeToByteArray()
 
-        return key.signJws(payload, jwtHeaders)
-    }
+    return key.signJws(payload, jwtHeaders)
+  }
 
-    @JvmBlocking
-    @JvmAsync
-    @JsPromise
-    @JsExport.Ignore
-    suspend fun verify(data: String): Result<JsonElement> = runCatching {
-        val jws = data.decodeJws()
+  @JvmBlocking
+  @JvmAsync
+  @JsPromise
+  @JsExport.Ignore
+  suspend fun verify(data: String): Result<JsonElement> = runCatching {
+    val jws = data.decodeJws()
 
-        val header = jws.header
-        val payload = jws.payload
+    val header = jws.header
+    val payload = jws.payload
 
-        val issuerDid = (payload[JwsOption.ISSUER] ?: header[JwsHeader.KEY_ID])!!.jsonPrimitive.content
-        log.trace { "Verifying with issuer did: $issuerDid" }
+    val issuerDid = (payload[JwsOption.ISSUER] ?: header[JwsHeader.KEY_ID])!!.jsonPrimitive.content
+    if (DidUtils.isDidUrl(issuerDid)) {
```

Comment:
maybe this could be extracted in a separate method, e.g.
```kotlin
private suspend fun verifyW3C(issuerDid: String, data: String): JsonElement {
    log.trace { "Verifying with issuer did: $issuerDid" }

    return DidService.resolveToKey(issuerDid)
      .also {
        if (log.isTraceEnabled()) {
          val exportedJwk = it.getOrNull()?.getPublicKey()?.exportJWK()
          log.trace { "Imported key: $it from did: $issuerDid, public is: $exportedJwk" }
        }
      }
      .getOrThrow()
      .verifyJws(data.split("~")[0])
      .also { log.trace { "Verification result: $it" } }
      .getOrThrow()
  }
```

and it becomes
```kotlin
if (DidUtils.isDidUrl(issuerDid)) {
  verifyW3C(issuerDid, data)
} else {
  TODO()
}
```